File: server/src/main/java/com/zzhgl/app/networking/response/room/ResponseJoinRoom.java
package com.zzhgl.app.networking.response.room;

import com.zzhgl.app.metadata.Constants;
import com.zzhgl.app.networking.response.GameResponse;
import com.zzhgl.app.utility.GamePacket;

public class ResponseJoinRoom extends GameResponse {
    private short status;
    private String roomId = "";
    private String roomName = "";

    public ResponseJoinRoom() {
        responseCode = Constants.SMSG_JOIN_ROOM;
    }

    @Override
    public byte[] constructResponseInBytes() {
        GamePacket packet = new GamePacket(responseCode);
        packet.addShort16(status);
        packet.addString(""); // Empty message
        if (status == Constants.SUCCESS) {
            packet.addString(roomId);
            packet.addString(roomName);
        }
        return packet.getBytes();
    }

    public void setStatus(short status) { this.status = status; }
    public void setRoomId(String roomId) { this.roomId = roomId; }
    public void setRoomName(String roomName) { this.roomName = roomName; }
}

File: client/Assets/Network/Room/Responses/ResponseJoinRoom.cs
using UnityEngine;

public class ResponseJoinRoomEventArgs : ExtendedEventArgs {
    public short Status { get; set; }
    public string Message { get; set; }
    public string RoomId { get; set; }
    public string RoomName { get; set; }

    public ResponseJoinRoomEventArgs() {
        Event_id = Constants.SMSG_JOIN_ROOM;
    }
}

public class ResponseJoinRoom : BaseNetworkResponse {
    private string roomId = "";
    private string roomName = "";

    public ResponseJoinRoom() {
        Response_id = Constants.SMSG_JOIN_ROOM;
    }
    
    protected override void ParseResponseData() {
        if (status == Constants.SUCCESS) {
            roomId = DataReader.ReadString(DataStream);
            roomName = DataReader.ReadString(DataStream);
        }
    }
    
    public override ExtendedEventArgs Process() {
        ResponseJoinRoomEventArgs args = new ResponseJoinRoomEventArgs();
        args.Status = status;
        args.Message = message;
        args.RoomId = roomId;
        args.RoomName = roomName;
        
        if (status == Constants.SUCCESS) {
            Constants.ROOM_ID = roomId;
            Constants.IS_HOST = false;
        }
        
        return args;
    }
}
